File: source/Emphasize/Format Definitions/Italic.vb
Imports Microsoft.VisualStudio.Text.Classification
Imports Microsoft.VisualStudio.Utilities
Imports System.ComponentModel.Composition


Namespace FormatDefinitions

    <ClassificationType(ClassificationTypeNames:=Italic.Name)>
    <Export(GetType(EditorFormatDefinition))>
    <Name(Italic.Name)>
    <Order(After:=Priority.High)>
    <UserVisible(True)>
    Public Class Italic
        Inherits ClassificationFormatDefinition


        Public Const Name As String = "Comment - Italic Span"


        <Export(GetType(ClassificationTypeDefinition))>
        <BaseDefinition("Comment")>
        <Name(Name)>
        Public Shared ReadOnly ClassificationType As ClassificationTypeDefinition


        Public Sub New()
            DisplayName = Name
            IsItalic = True
        End Sub

    End Class

End Namespace

File: source/Emphasize/FormatDefinitions/Italic.cs
using Microsoft.VisualStudio.Text.Classification;
using Microsoft.VisualStudio.Utilities;
using System.ComponentModel.Composition;


namespace Emphasize.FormatDefinitions {

    [ClassificationType(ClassificationTypeNames = Name)]
    [Export(typeof(EditorFormatDefinition))]
    [Name(Name)]
    [Order(After = Priority.High)]
    [UserVisible(true)]
    public class Italic : ClassificationFormatDefinition {

        public const string Name = "Emphasize - Italic";


        [Export(typeof(ClassificationTypeDefinition))]
        [Name(Name)]
        public static readonly ClassificationTypeDefinition ClassificationType;


        public Italic() {
            DisplayName = Name;
            IsItalic = true;
        }

    }

}
